### PR TITLE
realm: Fix crashing when .json file is invalid

### PR DESCRIPTION
**Pull request by backspace**
I encountered this problem while working on #687, having invalid JSON in a `.json` file produced this error on realm server startup:

```
There was an error running your app in fastboot. More info about the error:
 SyntaxError: Unexpected token I in JSON at position 0
    at JSON.parse (<anonymous>)
```

You can similarly [see it in that PR’s test output](https://github.com/cardstack/boxel/actions/runs/6331510555/job/17196212415?pr=687#step:7:137).

The failing test is [flapping on `main`](https://github.com/cardstack/boxel/actions/runs/6319481537/job/17160466505#step:7:311).

### CODE REVIEW

File: packages/host/app/lib/current-run.ts [hunk 1]
```diff
@@ -278,8 +278,16 @@ export class CurrentRun {
...
       if (url.href.endsWith('.json')) {
-        let { data: resource } = JSON.parse(content);
-        if (isCardResource(resource)) {
+        let resource;
+
+        try {
+          let { data } = JSON.parse(content);
+          resource = data;
+        } catch (e) {
+          log.debug(`unable to parse ${url.href} as card JSON`);
```
Comment (backspace) [inline packages/host/app/lib/current-run.ts]: This is a guess at how this should be handled, does something else need to happen if the parsing fails?

Comment (jurgenwerk) [reply packages/host/app/lib/current-run.ts]: I guess it would be good to actually tell the user? Maybe in some kind of an error display

**Review by jurgenwerk — APPROVED** [review]

File: packages/host/app/lib/current-run.ts (hunk 1 above)
Comment (backspace) [reply packages/host/app/lib/current-run.ts]: I’ve changed it to `warn`, this is something that happens with realm server indexing so it’s not directly user-facing. A related problem is that broken JSON silently (console-only) fails to save, the only way it’ll save is if it’s valid:

Boxel 2023-09-28 10-50-09

File: packages/host/app/lib/current-run.ts [hunk 2]
```diff
@@ -278,8 +278,16 @@ export class CurrentRun {
...
       if (url.href.endsWith('.json')) {
-        let { data: resource } = JSON.parse(content);
-        if (isCardResource(resource)) {
+        let resource;
+
+        try {
+          let { data } = JSON.parse(content);
+          resource = data;
+        } catch (e) {
+          log.warn(`unable to parse ${url.href} as card JSON`);
```
Comment (habdelra) [inline packages/host/app/lib/current-run.ts:287]: at this point we can't even say that it is card json. probably a better warning is 
```
log.warn(`unable to parse ${url.href} as JSON`);
```

**Pull request merged**